kore/test/Test/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -846,6 +848,98 @@ test_andTermsSimplification =
+                        & Condition.fromPredicate
+                actual <-
+                    simplifyEquals
+                        mempty
+                        ( Mock.builtinBool False )
+                        ( Mock.inKeysMap
+                            ( mkElemVar Mock.x)
+                            ( Mock.builtinMap
+                                [ ( mkElemVar Mock.y, Mock.a ) ]
+                            )
+                        )
+                assertEqual "" (Just [expect]) actual
+        , testCase
+            "\\equals(false, X in [Y Z])\
+            \ = \\not \\equals(X, Y) \\and \\not \\equals(X, Z)"
+            $ do
+                let expect =
+                        foldr1
+                            makeAndPredicate
+                            [ makeNotPredicate
+                                $ makeEqualsPredicate_
+                                    (mkElemVar Mock.x)
+                                    (mkElemVar Mock.y)
+                            , makeNotPredicate
+                                $ makeEqualsPredicate_
+                                    (mkElemVar Mock.x)
+                                    (mkElemVar Mock.z)
+                            -- Definedness condition
+                            , makeNotPredicate
+                                $ makeEqualsPredicate_
+                                    (mkElemVar Mock.y)
+                                    (mkElemVar Mock.z)
+                            ]
+                        & Condition.fromPredicate
+                actual <-
+                    simplifyEquals
+                        mempty
+                        (Mock.builtinBool False)
+                        ( Mock.inKeysMap
+                            ( mkElemVar Mock.x )
+                            ( Mock.builtinMap
+                                [ ( mkElemVar Mock.y, Mock.a )
+                                , ( mkElemVar Mock.z, Mock.a )
+                                ]
+                            )
+                        )
+                assertEqual "" (Just [expect]) actual
+        , testCase "\\equals(false, f(X) in [Y]) = \\not \\equals(f(X), Y)" $ do
+            let expect =
+                    makeAndPredicate
+                        ( makeNotPredicate
+                            ( makeAndPredicate
+                                ( makeCeilPredicate Mock.testSort
+                                    (Mock.f (mkElemVar Mock.x))
+                                )
+                                ( makeEqualsPredicate_
+                                    (mkElemVar Mock.y)
+                                    ( Mock.f (mkElemVar Mock.x) )
+                                )
+                            )
+                        )
+                        ( makeCeilPredicate Mock.testSort
+                            (Mock.f (mkElemVar Mock.x))
+                        )
+                    & Condition.fromPredicate
```

Comment:
This test passes, so I think the `actual` result is not completely simplified. Should I investigate more?

Reply:
You mean, it's not completely simplified because the `\ceil` appears twice?

Reply:
I though `(¬(ceil & eq)) & ceil` could be simplified to
`(¬ceil | ¬eq) & ceil`
`(¬ceil & ceil) | (¬eq & ceil)`
`¬eq & ceil`
which looks like the result you expected
`\equals(false, f(X) in [Y]) = \not \equals(f(X), Y) \and \ceil(f(X))`

Reply:
Yes, but this tests `termEquals`, which is not itself responsible for simplifying that condition.